### ISSUE TITLE
Fix 4 enumerateDevices

### ISSUE THREAD

**Issue by miroslavpejic85** [Dirvann/mediasoup-sfu-webrtc-video-rooms#12]
Hi @Dirvann, nice to meet you & 
Thanks for this useful project! ;) 

Small fix for `enumerateDevices`

https://github.com/Dirvann/mediasoup-sfu-webrtc-video-rooms/blob/master/public/index.js#L95

```js
function joinRoom(name, room_id) {
    if (rc && rc.isOpen()) {
        console.log('already connected to a room');
    } else {
        rc = new RoomClient(
            localMedia,
            remoteVideos,
            remoteAudios,
            window.mediasoupClient,
            socket,
            room_id,
            name,
            roomOpen,
        );

        addListeners();
        initEnumerateDevices(); // --->  call it from here when join in the room
    }
}

function initEnumerateDevices() {
   // Many browsers, without the consent of getUserMedia, cannot enumerate the devices. 

    const constraints = {
        audio: true,
        video: true,
    };

    navigator.mediaDevices
    .getUserMedia(constraints)
    .then(() => {
        enumerateDevices();
    })
    .catch((err) => {
        console.error('Access denied for audio/video: ', err);
    });
}

function enumerateDevices() {
    // Load mediaDevice options
    navigator.mediaDevices.enumerateDevices().then((devices) =>
        devices.forEach((device) => {
            let el = null;
            if ('audioinput' === device.kind) {
                el = audioSelect;
            } else if ('videoinput' === device.kind) {
                el = videoSelect;
            }
            if (!el) return;

            let option = document.createElement('option');
            option.value = device.deviceId;
            option.innerText = device.label;
            el.appendChild(option);
        }),
    );
}
```
In order to pass correctly, the `deviceId` to the `produce` instead of empty ;)
https://github.com/Dirvann/mediasoup-sfu-webrtc-video-rooms/blob/master/public/RoomClient.js#L250

**Comment by Dirvann** [comment]
Hi thanks for the improvement tip. I do think there is still a need to close the audio and video stream after getUsermedia otherwise it would keep them in use. Or is this not the case when you tested it?

**Comment by miroslavpejic85** [comment]
Yeah you right... :) you can close the  audio-video stream after getUsermedia...

**Comment by Dirvann** [comment]
Alright, I'll make the changes soon, unless you have a PR request ready, thanks!

**Comment by miroslavpejic85** [comment]
Just add this 
```js
    .then((stream) => {
        enumerateDevices();
        stream.getTracks().forEach(function(track) {
            track.stop();
        });
    })
```

**Comment by miroslavpejic85** [comment]
> Alright, I'll make the changes soon, unless you have a PR request ready, thanks!
As you prefer, I can also get you a PR if you want? My pleasure to help ;)

**Comment by Dirvann** [comment]
That would be awesome!

**Comment by miroslavpejic85** [comment]
PS: If you have pleasure, we can use [prettier](https://github.com/prettier/prettier) As a code style, more readable... I can do it in the next PR ;)

**Comment by Dirvann** [comment]
solved in #13
